src/webnn/native/dml/dml_platform.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 The WebNN-native Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef WEBNN_NATIVE_DML_DMLPLATFORM_H_
```

Comment:
According to the [Header Files](https://google.github.io/styleguide/cppguide.html#Header_Files) in google c++ style guilde, should we remove this dml_platform.h to include these header files where they are needed? @bbernhar

Reply:
I would move `WEBNN_RETURN_IF_FAILED` into `ErrorDML.h` since it's code does not depend on any #include preceding it.

It's nice to have like-includes (depend on each other) in one place rather then in multiple `.cpp` files - I don't think this goes against the style guide recommendations.

Reply:
OK. Will create ErrorDML.h if there are more codes related to error handling.

Reply:
SGTM.